File: server-laravel/app/Http/Controllers/Produit.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

// Modèle Produit — MANANJAFY Andreas Aslain
// Projet 4 Catalogue Produits — Groupe M&A

class Produit extends Model
{
    // champs qu'on peut remplir depuis le controller
    protected $fillable = [
        'nom',
        'description',
        'prix',
        'stock',
        'reference',
    ];

    protected $casts = [
        'prix'  => 'float',
        'stock' => 'integer',
    ];
}

File: app/src/main/java/com/ma/catalogueproduits/model/Produit.kt
package com.ma.catalogueproduits.model

// Data class représentant un produit
// MANANJAFY Andreas Aslain — Groupe M&A

data class Produit(
    val id: Int = 0,
    val nom: String = "",
    val description: String = "",
    val prix: Double = 0.0,
    val stock: Int = 0,
    val reference: String = ""
)
